front/lib/resources/vault_resource.ts

```diff
@@ -159,18 +163,22 @@ export class VaultResource extends BaseResource<VaultModel> {
       return null;
     }
 
-    const vault = await this.model.findOne({
+    const vaultModel = await this.model.findOne({
       where: {
         id: vaultModelId,
         workspaceId: owner.id,
       },
     });
 
-    if (!vault) {
+    if (!vaultModel) {
+      return null;
+    }
+    const vault = new this(VaultModel, vaultModel.get());
+    if (!auth.isAdmin() && !auth.hasPermission([vault.acl()], "read")) {
```

Comment:
I thought we agreed on adding this permission layer only on the list?

Reply:
right .. forgot to remove that one